src/mlir/IR/Attribute.jl

```diff
@@ -504,118 +528,118 @@ Creates a dense elements attribute with the given shaped type from elements of a
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{UInt32})
     shaped_type = TensorType(size(values), Type(UInt32))
-    return Attribute(
+    return DenseElementsAttribute{UInt32}(Attribute(
         API.mlirDenseElementsAttrUInt32Get(
             shaped_type, length(values), to_row_major(values)
         ),
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{Int32})
     shaped_type = TensorType(size(values), Type(Int32))
-    return Attribute(
+    return DenseElementsAttribute{Int32}(Attribute(
         API.mlirDenseElementsAttrInt32Get(shaped_type, length(values), to_row_major(values))
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{UInt64})
     shaped_type = TensorType(size(values), Type(UInt64))
-    return Attribute(
+    return DenseElementsAttribute{UInt64}(Attribute(
         API.mlirDenseElementsAttrUInt64Get(
             shaped_type, length(values), to_row_major(values)
         ),
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{Int64})
     shaped_type = TensorType(size(values), Type(Int64))
-    return Attribute(
+    return DenseElementsAttribute{Int64}(Attribute(
         API.mlirDenseElementsAttrInt64Get(shaped_type, length(values), to_row_major(values))
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{Float32})
     shaped_type = TensorType(size(values), Type(Float32))
-    return Attribute(
+    return DenseElementsAttribute{Float32}(Attribute(
         API.mlirDenseElementsAttrFloatGet(shaped_type, length(values), to_row_major(values))
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray{Float64})
     shaped_type = TensorType(size(values), Type(Float64))
-    return Attribute(
+    return DenseElementsAttribute{Float64}(Attribute(
         API.mlirDenseElementsAttrDoubleGet(
             shaped_type, length(values), to_row_major(values)
         ),
-    )
+    ))
 end
 
 if isdefined(Core, :BFloat16)
     function DenseElementsAttribute(values::AbstractArray{Core.BFloat16})
         shaped_type = TensorType(size(values), Type(Core.BFloat16))
-        return Attribute(
+        return DenseElementsAttribute{Core.BFloat16}(Attribute(
             API.mlirDenseElementsAttrBFloat16Get(
                 shaped_type, length(values), to_row_major(values)
             ),
-        )
+        ))
     end
 end
 
 function DenseElementsAttribute(values::AbstractArray{Float16})
     shaped_type = TensorType(size(values), Type(Float16))
-    return Attribute(
+    return DenseElementsAttribute{Float16}(Attribute(
         API.mlirDenseElementsAttrFloat16Get(
             shaped_type, length(values), to_row_major(values)
         ),
-    )
+    ))
 end
 
 function DenseElementsAttribute(values::AbstractArray)
     shaped_type = TensorType(size(values), Type(eltype(values)))
-    return Attribute(
+    return DenseElementsAttribute{:unknown}(Attribute(
```

Comment:
mmm i think it may be better to put here `eltype(values)`

```suggestion
    return DenseElementsAttribute{eltype(values)}(Attribute(
```